sjb/config/test_cases/ci-kubernetes-machine-api-operator.yml

```diff
@@ -0,0 +1,232 @@
+---
+parent: 'common/test_cases/minimal.yml'
+extensions:
+  actions:
+    - type: "host_script"
+      title: "Upload the default AWS credentiasl"
+      script: |-
+         ssh -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config openshiftdevel "mkdir ~/.aws"
+         scp -F ${WORKSPACE}/.config/origin-ci-tool/inventory/.ssh_config ~/.aws/credentials openshiftdevel:~/.aws
+    - type: "script"
+      title: "Install minikube"
+      script: |-
+        # Install minikube
+        curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.28.0/minikube-linux-amd64
+        chmod +x minikube
+        sudo mv minikube /usr/bin/
+        # Install kubectl
+        curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl
+        chmod +x kubectl
+        sudo mv kubectl /usr/bin/
+    - type: "script"
+      title: "Deploy kubernetes"
+      script: |-
+        sudo setenforce 0
+        # https://github.com/kubernetes/minikube/blob/master/docs/configuring_kubernetes.md
+        sudo minikube start --vm-driver=none --extra-config=kubelet.cgroup-driver=systemd
+    - type: "script"
+      title: "Build the machine api operator"
+      script: |-
+        # build the mao image
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/machine-api-operator
+        go build -v -o bin/machine-api-operator github.com/openshift/machine-api-operator/cmd
+        go get -u github.com/openshift/imagebuilder/cmd/imagebuilder
+        sudo mv /data/bin/imagebuilder /usr/bin
+        sudo imagebuilder -t "docker.io/machine-api-operator:$(git describe --always --abbrev=7)" .
+    - type: "script"
+      title: "Deploy the machine api operator"
+      script: |-
+        export GOPATH=/data
+        cd $GOPATH/src/github.com/openshift/machine-api-operator
+
+        # Drop the tectonic-system ns
+        cat << HEREDOC > namespace.yaml
```

Comment:
We need this also for local testing (it is also mentioned in README), so cannot we create this file in machine-api-operator repo, for example in `tests` directory and just reuse it here?

Reply:
That's the whole idea. We change file, CI uses it to test what we are doing.

Reply:
Yeah, would like to see as much as possible in the repo under test itself, so that the test is useful outside of CI as well. In the future (VM jobs without aos-cd-jobs sjb stuff) we are going to heavily favor jobs that run a single `make` target or script from the repo under test.